src/ol-ext/format/featurehash.js

```diff
@@ -0,0 +1,940 @@
+ *   can define attribute and style transformers. This is currently
+ *   not supported by the ngeo format.
+ */
+goog.provide('ngeo.format.FeatureHash');
+
+goog.require('goog.asserts');
+goog.require('ol.Feature');
+goog.require('ol.format.TextFeature');
+goog.require('ol.geom.GeometryLayout');
+goog.require('ol.geom.GeometryType');
+goog.require('ol.geom.LineString');
+goog.require('ol.geom.MultiLineString');
+goog.require('ol.geom.MultiPoint');
+goog.require('ol.geom.MultiPolygon');
+goog.require('ol.geom.Point');
+goog.require('ol.geom.Polygon');
+goog.require('ol.style.Circle');
+goog.require('ol.style.Fill');
+goog.require('ol.style.Stroke');
+goog.require('ol.style.Style');
+goog.require('ol.style.Text');
+
+
+/**
+ * @enum {string}
+ */
+ngeo.format.FeatureHashStyleType = {
+  LINE_STRING: 'LineString',
+  POINT: 'Point',
+  POLYGON: 'Polygon'
+};
+
+
+/**
+ * @type {Object.<ol.geom.GeometryType, ngeo.format.FeatureHashStyleType>}
+ * @private
+ */
+ngeo.format.FeatureHashStyleTypes_ = {};
+
+ngeo.format.FeatureHashStyleTypes_[ol.geom.GeometryType.LINE_STRING] =
+    ngeo.format.FeatureHashStyleType.LINE_STRING;
+ngeo.format.FeatureHashStyleTypes_[ol.geom.GeometryType.POINT] =
+    ngeo.format.FeatureHashStyleType.POINT;
+ngeo.format.FeatureHashStyleTypes_[ol.geom.GeometryType.POLYGON] =
+    ngeo.format.FeatureHashStyleType.POLYGON;
+ngeo.format.FeatureHashStyleTypes_[ol.geom.GeometryType.MULTI_LINE_STRING] =
+    ngeo.format.FeatureHashStyleType.LINE_STRING;
+ngeo.format.FeatureHashStyleTypes_[ol.geom.GeometryType.MULTI_POINT] =
+    ngeo.format.FeatureHashStyleType.POINT;
+ngeo.format.FeatureHashStyleTypes_[ol.geom.GeometryType.MULTI_POLYGON] =
+    ngeo.format.FeatureHashStyleType.POLYGON;
+
+
+
+/**
+ * @constructor
+ * @param {ngeox.format.FeatureHashOptions=} opt_options Options.
+ * @extends {ol.format.TextFeature}
+ */
+ngeo.format.FeatureHash = function(opt_options) {
+  goog.base(this);
+
+  var options = goog.isDef(opt_options) ? opt_options : {};
+
+  /**
+   * @type {number}
+   * @private
+   */
+  this.accuracy_ = goog.isDef(options.accuracy) ?
+      options.accuracy : ngeo.format.FeatureHash.ACCURACY_;
+
+  /**
+   * @type {number}
+   * @private
+   */
+  this.prevX_ = 0;
+
+  /**
+   * @type {number}
+   * @private
+   */
+  this.prevY_ = 0;
+
+};
+goog.inherits(ngeo.format.FeatureHash, ol.format.TextFeature);
+
+
+/**
+ * Characters used to encode the data.
+ * ~'() will be used as separator.
+ * @const
+ * @private
+ */
+ngeo.format.FeatureHash.CHAR64_ =
+    '.-_!*ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789abcdefghjkmnpqrstuvwxyz';
+
+
+/**
+ * @const
+ * @private
+ */
+ngeo.format.FeatureHash.ACCURACY_ = 1;
+
+
+/**
+ * @param {number} num Number.
+ * @return {string} String.
+ * @private
+ */
+ngeo.format.FeatureHash.encodeSignedNumber_ = function(num) {
+  var sgnNum = num << 1;
+  if (num < 0) {
+    sgnNum = ~(sgnNum);
+  }
+  return ngeo.format.FeatureHash.encodeNumber_(sgnNum);
+};
```

Comment:
I'll rename it to `signedNum` because the argument name is `num` and I don't want to rename `num` to `number` as `number` is reserved word.